### PR TITLE
Introduce Begin/EndComputePass

### PR DESCRIPTION
**Pull request by kainino0x**
`SetPipeline(computePipeline)` is gated on `BeginComputePass`, so that should cover the necessary validation, at least for `Dispatch`.

### CODE REVIEW

File: src/backend/metal/CommandBufferMTL.mm [hunk 1]
```diff
@@ -141,6 +141,13 @@ void EndRenderPass() {
         uint32_t currentSubpass = 0;
         while (commands.NextCommandId(&type)) {
             switch (type) {
+                case Command::BeginComputePass:
+                    {
+                        commands.NextCommand<BeginComputePassCmd>();
+                        // XXX: begin compute pass
```
Comment (Kangz) [inline src/backend/metal/CommandBufferMTL.mm]: Can you make it a TODO(kainino) ... here and below?

File: src/backend/CommandBufferStateTracker.cpp [hunk 2]
```diff
@@ -61,7 +61,7 @@ namespace backend {
 
     bool CommandBufferStateTracker::ValidateCanDispatch() {
```
Comment (Kangz) [inline src/backend/CommandBufferStateTracker.cpp]: We should check that we aren't in a compute pass at the end of ValidateGetResult.

File: src/backend/metal/CommandBufferMTL.mm (hunk 1 above)
Comment (kainino0x) [inline src/backend/metal/CommandBufferMTL.mm]: Oh yeah. Actually, let me see if I can just DO this TODO.

File: src/backend/CommandBufferStateTracker.cpp (hunk 2 above)
Comment (kainino0x) [inline src/backend/CommandBufferStateTracker.cpp]: Done

**Comment by kainino0x** [comment]
Done, could you TAL at the last commit?

**Pull request merged**